components/director/internal/domain/fetchrequest/service.go

```diff
@@ -0,0 +1,74 @@
+package fetchrequest
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/internal/timestamp"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+)
+
+type service struct {
+	client       *http.Client
+	logger       *log.Logger
+	timestampGen timestamp.Generator
+}
+
+//go:generate mockery -name=FetchRequestRepository -output=automock -outpkg=automock -case=underscore
+type FetchRequestRepository interface {
+	Update(ctx context.Context, item *model.FetchRequest) error
+}
+
+func NewService(client *http.Client, logger *log.Logger) *service {
+	return &service{
+		client:       client,
+		logger:       logger,
+		timestampGen: timestamp.DefaultGenerator(),
+	}
+}
+
+func (s *service) SetTimestampGen(timestampGen func() time.Time) {
```

Comment:
This function should be in `export_test.go` file. I see that you wrote a similar function in status update related code, please move that one to `export_test.go` as well.